### PR TITLE
feat: added health details and medical history fields to consultation form

### LINKED ISSUE #2738: Changes to be made in the consultation form : Add Health Details
Part of #2218 
HEALTH DETAILS

 - Add a new collapsable section to the consultation form on top. This section will contain all health details of the Patient.
 - The first field is "Allergy"- YES/NO. If YES, a short text box for details. (the text field already exists in the patient registration form. the same may be migrated)
 - Second field will be for "Vaccinations". The user must be able to select multe inputs from the options. The Options would be: COVID (1 dose, 2 dose), Hep B, Hep A. (The field for COVID Vaccinationa already exists in patient registration form)
 - "Blood Group" With dropdown. (The field already exists in the patient registration form and may be migrated here)
 - Field for "Weight" and "Height" existing at the bottom of the consultation form may be moved up to the top after blood group.
 - Add a Long Text Field for "Familty Details"

### LINKED ISSUE #3331: Move "Health Details" and "Medical History" from patient registration form to consultation form
Move the "Health Details" and "Medical History"
From 
patient registration form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/update)
to 
Consultation Form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/consultation)

-The sections must come on top as collapsable sections
- The sections must come expanded only in the first consultation on a patient card in care. For all subsequent consultations, the sections must be minimized and only the header must be visible, with the option to expand.
- The sections will be blank only for the first consultation for a patient in care. For every subsequent consultation, the sections must come prefilled with previous consultation data. The user must be able to edit and save.
- The existing content in the consultation form will come under the section heading "consultation"
- "Medical History" to appear on top, then "Health Details" and last place the "Consultation"
- A suggestive design is as under (DONOT follow the design for details. Follow the existing styling, fonts etc in CARE)
Screenshot 2022-08-05 at 9 42 01 PM

@gigincg Please review
Part of #2218

### LINKED ISSUE #3332: Allergies: to be added to health details
Note: #3331 moves the health details and medical history section from the patient registration form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/update) to Consultation Form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/consultation)

- [ ] Health details must contain a toggle button (Yes/NO) for "Allergies".  Default selection to be NO

- [ ] If selected "Yes"- a short text box must appear for description.
NOTE: A short text box for allergies already exists in the Medical History section. The same may be moved and used here. 

This shall be placed as the first field in Health details

- [ ] Wherever allergy is selected to YES, make the allergy details appear on patient dashboard (https://care.coronasafe.in/facility/015ad911-39b7-4104-b304-bf8178c33830/patient/76742faa-30dd-4eb9-9649-fe718f947ec0/consultation/2ad89b7c-7bff-4035-a3c6-36d53551a8f0) below diagnosis.

- [ ] Create a label for allergies to appear in the patient list

@gigincg to review
Issue is part of #2218

### LINKED ISSUE #3333: Vaccinations: in Health details
Note: https://github.com/coronasafe/care_fe/issues/3331 moves the health details and medical history section from the patient registration form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/update) to Consultation Form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/consultation)

Add a section for vaccinations. 

- [ ] The first field would be to select the type of Vaccine. The options would be "COVID", "Hep A"  "Hep B". The options may be expanded later. The user should be able to select multiple options. 
- [ ] Once a vaccine in selected, there must be the option for a dose for certain options. In this case for COVID the option of doses are going to be 1,2,and 3. For the other options, the field for dosage will not appear.
- [ ] Date of the last dose- this must appear for all options. The field for the date must accept the exact date (DD/MM/YYYY), Month and year (MM/YYYY) or even just year (YYYY)

The design of the field may mimic the prescription module where one vaccine may be entered along with details, and another row may be added to add a second vaccine. This is only a suggestion, you may use your creativity to come up with better layouts

- [ ] The information thus collected must be shown in the Patient Details Page (https://care.coronasafe.in/facility/015ad911-39b7-4104-b304-bf8178c33830/patient/aefb147a-eb09-4dd0-8390-e2cd5a05f326) on the right side of Bloof Group below the Patient Name.

NOTE: 5 fields relating to COVID vaccines already appear in health details. The field are "Is the patient vaccinated against COVID?" "COWIN ID", "No. of doses", "Vaccine Name" and "last date of vaccination". These fields may be repurposed/ removed with this new vaccine section that covers a wider range of vaccines.

@gigincg to review
This is part of #2218

### LINKED ISSUE #3335: Blood Group: reposition
Note: https://github.com/coronasafe/care_fe/issues/3331 moves the health details and medical history section from the patient registration form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/update) to Consultation Form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/consultation)

The blood group field with dropdown is currently under the "Medical History" Section. 
It must be moved to the "health details" section

@gigincg Please review
Part of #2218

### LINKED ISSUE #3336: Height and Weight : Reposition
Note: https://github.com/coronasafe/care_fe/issues/3331 moves the health details and medical history section from the patient registration form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/update) to Consultation Form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/consultation)

The fields for Height and Weight fields exist in consultation.

Move the same to "Health Details"

Part of #2218 
@gigincg please review

### LINKED ISSUE #3338: Family Details- add field in health details
Note: https://github.com/coronasafe/care_fe/issues/3331 moves the health details and medical history section from the patient registration form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/update) to Consultation Form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/consultation)

- [ ] As the last field in "Health Details" add a long text field for "Family Details"
The field will be non-mandatory

- [ ] Show the information in Patient details page (https://care.coronasafe.in/facility/015ad911-39b7-4104-b304-bf8178c33830/patient/76742faa-30dd-4eb9-9649-fe718f947ec0) as a separate heading "Family Details" below "Medical"
Screenshot 2022-08-06 at 12 59 03 AM

Issue is part of #2218 
@gigincg please review

### LINKED ISSUE #3361: Medical History: Introduce timeline format
Note: https://github.com/coronasafe/care_fe/issues/3331 moves the health details and medical history section from the patient registration form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/update) to Consultation Form (https://care.coronasafe.in/facility/cae151a7-1b0d-4d76-9d61-a2733b65c648/patient/69b26ab0-a158-414f-950a-f033ea859e51/consultation)

In the Medical History section, Retain the "Present Health Condition" and "Ongoing Medication field" as it is.  The Blood group and Allergies field are moved to the health details through #3332 and #3335. 

There is a field "Any medical history? " with multiple choice options. 

Replace this with the following. 

DATA INPUT

- [ ] Section heading: Medical History (this will be a sub-section within the larger medical history collapsable section itself) refer to the design below

- [ ] A field on the left will capture Date as DD/MM/YYYY. The user needs to have the option of choosing only Month and year (MM/YYYY) or only the Year (YYYY).

- [ ] Once the date is selected in the middle, a text field to search through all diseases must appear. This text field must allow the user to type and search through any disease in ICD-11 (Internal Classification of diseases available at https://icd.who.int/en). The user must be able to find the right disease and select it. Enable multi Select
NOTE: ICD-11 is integrated into the "Diagnosis" field in CARE is #2016 

- [ ] Once the disease is selected, a long text description box must also appear.

- [ ] An option to add more sections of date, ICD and description must always appear below (Same design as the prescription module). This will allow users to keep adding more and more details about their medical history across various dates. 

DATA VISUALISATION 

- [ ] Once saved, fields will appear in Chronological order 

The design to be followed is indicated in the below design
Screenshot 2022-08-09 at 1 31 26 PM

The larger "Medical History" section itself will only come expanded in the first consultation of a patient in CARE. In all other subsequent consultations, the section will appear minimized in the consultation form and will expand to show the content only when the user clicks on the header. 

- [ ] Additional Feature (may be separated out as a separate issue)
While viewing saved medical history, For every saved date, ICD and description, there needs to be an option on the right-hand side to "mark as cured". This will allow the system to not only list all the diseases the patient has but also understand which of them is already cured.

Part of #2218
@gigincg please review

### PR DESCRIPTION
**Pull request by cp-Coder**
# Proposed Changes

- Adds Health details and medical history section in the consultation form

## Associated Issue

- Fixes #2738
- Fixes #3331 
- Fixes #3332
- Fixes #3333
- Fixes #3335
- Fixes #3336
- Fixes #3338
- Fixes #3361

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

### CODE REVIEW

**Comment by nihal467** [comment]
@cp-Coder marking it for team discussion required, for reevaluating the PR since the backend changes are not getting merged, and the issue seems to be changed a lot,

**Comment by nihal467** [comment]
@rithviknishad can you review it @cp-Coder can you work on the backend

**Comment by rithviknishad** [comment]
@cp-Coder could you clear the merge conflicts too?

**Comment by rithviknishad** [comment]
@cp-Coder could you clear the merge conflicts?

**Review by rithviknishad — CHANGES_REQUESTED** [review]
@cp-Coder @nihal467 few issues that are to be solved and few minor issues which could be taken up later.

### Issues
- [x] Nothing happens when clicking add patient in Patient Registration form. Unable to create a patient even with all fields in Personal Details section filled, COVID details were never opened. However, even after filling all COVID details fields, no errors seems to appear, nor it creates the patient. No console errors, no network activity.
image


### Other minor issues (optional, but could be taken up as seperate issues even)
- [ ] Part of the Patient Registration form is back to using old components. Hence errors are in black instead of red.
- [ ] **MISSING**: Create a label for allergies to appear in the patient list - _part of #3332_

**Review by rithviknishad — CHANGES_REQUESTED** [review]
(ignore, I forgot to pull latest backend, my bad)

~@cp-Coder unable to create a consultation now. Getting 500 internal server error in your backend branch.~

~cc: @nihal467~

image

**Comment by rithviknishad** [comment]
We could take this up as a separate issue

image

**Review by rithviknishad — APPROVED** [review]
Looks okay :) @nihal467

**Comment by nihal467** [comment]
@cp-Coder the deploy is failed and there is a merge conflict, can you clear it ASAP

File: src/Components/Facility/ConsultationForm.tsx [hunk 1]
```diff
@@ -519,7 +591,65 @@ export const ConsultationForm = (props: any) => {
         case "verified_by": {
```
Comment (rithviknishad) [inline src/Components/Facility/ConsultationForm.tsx]:
```suggestion
        case "verified_by":
```

**Comment by nihal467** [comment]
LGTM, @khavinshankar merge it ASAP, we are having a lot of minor bugs in the PR, which are not causing functionality issues but need to be fixed this week itself, so once this PR is merged, the following issues will be filed

**Comment by nihal467** [comment]
@cp-Coder 
image

- [x] when i scroll down, and reach the health section, it is still showing I am in the general details on the left side
- [x] rename "General Details" to "Consultation"
- [x] As per issue #3331 rearrange the order from consultation-health details-medical history to medical history-health details-consultation
- [x] as per issue #3338 move the family details field to the last of the health details section ( under the weight and height field )
- [x] as per issue [#2218](https://github.com/coronasafe/care_fe/issues/2218) reorder the health detail section as allergy - vaccination - blood group - weight & height - family details

image

- [x] as per issue [#3361](https://github.com/coronasafe/care_fe/issues/3361) , rename disease history to medical history
- [ ] it should be appearing in chronological order, once we submit the data

**Comment by aeswibon** [comment]
> - [ ] it should be appearing in chronological order, once we submit the data

@nihal467 Can you elaborate it more? What should be in the chronological order?

**Comment by khavinshankar** [comment]
@nihal467 Is this pr relevant now?

@cp-Coder what's the status of this pr?

**Comment by aeswibon** [comment]
> @nihal467 Is this pr relevant now?
>
> @cp-Coder what's the status of this pr?          

@khavinshankar The PR is good for testing.
The PR has been tested before and @nihal467 added the `tested` label before. but due to merge conflicts, he suggested some additional changes(minor UI changes). Just waiting for @nihal467 to test it once again.

**Comment by nihal467** [comment]
@cp-Coder due to large merge conflict and inactivity in the PR, closing this PR

**Pull request closed without merging**